### PR TITLE
Follows spec w3 c

### PR DESCRIPTION
**Pull request by maverickmishra**
### Platforms affected
iOS, Android, Windows
### What does this PR do?
 The PR makes the plugin conform to exact W3C specs. screen.orientation.angle and screen.orientation.onchange attributes implemented.

### What testing has been done on this change?
Tested on android and iOS devices and also using unit tests.

### Checklist
- [ ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [ ] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by cordova-qa** [comment]
Cordova CI Build has one or more failures. 

**Commit**     - [Link](https://github.com/apache/cordova-plugin-screen-orientation/pull/12/commits/60f639b5538a53ce056f14e2a70894247126b279)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=windows-10-store/artifact/) |
| [iOS 9.3]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-9.3/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-9.3/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-9.3/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-9.3/artifact/) |
| [iOS 10.0]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-10.0/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-10.0/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-10.0/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=ios-10.0/artifact/) |
| [Android 4.4]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-4.4/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-4.4/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-4.4/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-4.4/artifact/) |
| [Android 5.1]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-5.1/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-5.1/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-5.1/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/18//PLATFORM=android-5.1/artifact/) |

File: demo/config.xml [hunk 1]
```diff
@@ -1,14 +1,31 @@
+/*
```
Comment (purplecabbage) [inline demo/config.xml]: This is suddenly NOT a valid XML file ...
See: https://github.com/apache/cordova-plugin-screen-orientation/blob/master/plugin.xml#L1 for an example of how to add the license header to XML files.

File: www/screenorientation.js [hunk 2]
```diff
@@ -18,83 +18,127 @@
...
+}
+
+addScreenOrientationApi(screen.orientation);
+
+var onChangeListener = null;
+Object.defineProperty(screen.orientation, 'onchange', {
+
+    set: function(listener) {
+        if (onChangeListener != null) {
+            screen.orienation.removeEventListener('change', onChangeListener);
```
Comment (purplecabbage) [inline www/screenorientation.js]: Spelling error: orienation <= orientation

**Comment by cordova-qa** [comment]
Cordova CI Build has one or more failures. 

**Commit**     - [Link](https://github.com/apache/cordova-plugin-screen-orientation/pull/12/commits/73d48d47c9f374c5063bda734398af1f03b0e41c)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=windows-10-store/artifact/) |
| [iOS 9.3]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-9.3/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-9.3/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-9.3/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-9.3/artifact/) |
| [iOS 10.0]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-10.0/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-10.0/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-10.0/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=ios-10.0/artifact/) |
| [Android 4.4]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-4.4/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-4.4/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-4.4/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-4.4/artifact/) |
| [Android 5.1]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-5.1/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-5.1/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-5.1/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-screen-orientation-pr/19//PLATFORM=android-5.1/artifact/) |

**Pull request closed without merging**